### DISCUSSION TITLE
Why I use v = sdpvar(n,n,'symmetric') in yalmip but get a non-symmetric solution?

### DISCUSSION THREAD

**Discussion by KarryMa** [yalmip/YALMIP#1189] [Problems using YALMIP]
Dear all,
I define variables v = sdpvar(n,n) in the yalmip, 
` Constraints = [v(2,2)+v(4,1)*2^n==1;v(1,1)==v(2,2)+v(4,1);v>=0];`
My solution just gives me 
![52001e229267e4eb914a5a5473f52cd](https://user-images.githubusercontent.com/53418276/197141132-672ac7bd-6ba4-40b4-936f-932c6d297d6a.jpg)
But I hope I can get 
![image](https://user-images.githubusercontent.com/53418276/197140244-147c1c0c-4649-4000-81fe-8feff5d2f094.png)
How can I define a variable like this? Does the matrix have a name? I know I can add more constraints, but I don't want to do that.

**Comment by johanlofberg** [comment]
Looks symmetric to me, so not clear what you are talking about. Symmetric is default, so no reason to use the 'symmetric' flag

**Reply by KarryMa** [reply]
Yes, you are right I changed my question.

**Reply by johanlofberg** [reply]
You hope you can get? Nothing in the constraints tells the solver you want that particular solution, or a solution with that structure, not have you constructed a matrix with that structure

`X = diag(sdpvar(n,1));y = sdpvar(1);X(1,end)=y;X(end,1)=y;`

**Reply by KarryMa** [reply]
It seems that the upper triangular matrix is equal to the lower triangular matrix of this matrix, does the kind of matrix have a name?

**Reply by johanlofberg** [reply]
That sounds like the definition of a symmetric matrix
